python/its-info/its_info/main.py

```diff
@@ -0,0 +1,127 @@
+        self.ntp_host = cfg.get("general", "ntp_host", fallback=None)
+        self.interface = cfg.get("general", "interface", fallback=None)
+
+        self.client_id = cfg.get("mqtt", "client_id", fallback=self.instance_id)
+        self.host = cfg.get("mqtt", "host", fallback="127.0.0.1")
+        self.port = cfg.getint("mqtt", "port", fallback=1883)
+        self.username = cfg.get("mqtt", "username", fallback=None)
+        self.password = cfg.get("mqtt", "password", fallback=None)
+        self.topic = cfg.get("mqtt", "topic", fallback="info")
+        self.retry = cfg.getint("mqtt", "retry", fallback=2)
+
+        self.client = paho.mqtt.client.Client(
+            client_id=self.client_id,
+        )
+        self.client.reconnect_delay_set(min_delay=1, max_delay=self.retry)
+        self.client.username_pw_set(self.username, self.password)
+        self.client.on_connect = self.on_connect
+        self.client.on_disconnect = self.on_disconnect
+        self.client.on_socket_close = self.on_socket_close
+        self.client.connect_async(host=self.host, port=self.port)
+
+        self.timer = linuxfd.timerfd(closeOnExec=True)
+
+    def loop_forever(self):
+        self.client.loop_start()
+        while True:
+            try:
+                # We don't care about the number of time the timer has
+                # fired that we missed; given the periodicity is in the
+                # order of many seconds, it is highly unlikely that we
+                # ever miss a tick...
+                self.timer.read()
+                self.info()
+            except InterruptedError:
+                # Someone sent a signal to this thread...
+                continue
+            except KeyboardInterrupt:
+                break
+
+        self.client.loop_stop()
+        self.client.disconnect()
+
+    def on_connect(self, _client, _userdata, _flags, _rc, _properties=None):
+        self.ips = list()
+        if self.interface is not None:
+            ifa = netifaces.ifaddresses(self.interface)
+            for familly in [netifaces.AF_INET, netifaces.AF_INET6]:
+                if fammilly in ifa:
+                    # IPv6 addresses can look like: 01::EF%iface but we just want 01::EF
+                    self.ips += [i["addr"].split("%")[0] for i in ifa[familly]]
+        self.timer.settime(value=self.period, interval=self.period)
+        self.info()
+
+    def on_disconnect(self, _client, _userdata, _rc, _properties=None):
+        self.timer.settime(value=0)
+
+    def on_socket_close(self, _client, _userdata, _sock):
+        self.timer.settime(value=0)
+
+    def info(self):
+        data = {
+            "type": "broker",
+            "version": "1.1.0",
+            "instance_id": self.instance_id,
+            "instance_type": "edge",
+            "running": True,
+            "timestamp": int(1000 * time.time()),
+            "validity_duration": int(self.period * 2),
```

Comment:
The period is set in seconds in the config file, and retrieved as such in `self.period`, while `validity_duration` should be in milliseconds: to be multiplied by 2000 instead of 2